libr/core/cmd.c

```diff
@@ -4007,20 +4022,27 @@ R_API int r_core_cmd_foreach3(RCore *core, const char *cmd, char *each) { // "@@
 		}
 #endif
 		break;
-	case 's':
+	case 's': // XXX this command will crash when used with 'oc' (ocm'@@@s)
 		if (each[1] == 't') { // strings
 			list = r_bin_get_strings (core->bin);
 			RBinString *s;
 			if (list) {
 				ut64 offorig = core->offset;
 				ut64 obs = core->blocksize;
+				RBinString *s;
+				RList *lost = r_list_newf (free);
 				r_list_foreach (list, iter, s) {
+					RBinString *bs = r_mem_dup (s, sizeof (RBinString));
```

Comment:
it would be so cool to have refcounting in these cases :(

Reply:
yes it will